### DISCUSSION TITLE
Implementing Autogen Sequential Workflow approach in FastAPI and doesn't works.

### DISCUSSION THREAD

**Discussion by IvanCord** [microsoft/autogen#4823] [Q&A] [answered]
Hello folks!

I’m learning Autogen and experimenting with a small project that uses the Sequential Workflow approach (referencing [the official docs](https://microsoft.github.io/autogen/0.4.0.dev11/user-guide/core-user-guide/design-patterns/sequential-workflow.html)) to make three agents communicate in this order:
	1.	ValidatorAgent
	2.	ConversatorAgent
	3.	UserAgent

I’m using a pub-sub approach, and the final UserAgent acts as a closure agent that queues the final result to a service endpoint.

While implementing this in FastAPI, I always receive a 400 Bad Request error. In my run_workflow function, I compile the agents and see the log message “ValidatorAgent initialized,” but never any log from the ConversatorAgent—so it appears the pipeline doesn’t progress to that point before publishing the message.

I’ve tried debugging and ensuring my implementation is correct, but I can’t spot the mistake. Any help would be appreciated.

![image](https://github.com/user-attachments/assets/4821cbfc-ee51-40c2-a086-92df1450e08c)
![image](https://github.com/user-attachments/assets/73b4ee4a-65d6-4837-b182-55596fbfecbe)
![image](https://github.com/user-attachments/assets/9b59ffcc-eab3-40e4-bc5e-9ea73d5d0568)
![image](https://github.com/user-attachments/assets/a751e435-8f23-41c1-a6d6-e796cb401a7a)
![image](https://github.com/user-attachments/assets/cc60065a-80e2-4a41-b950-71e663a06e3c)
![image](https://github.com/user-attachments/assets/475ddc88-423e-4ed4-95cc-eca01669b676)
![image](https://github.com/user-attachments/assets/8a6ee434-280c-4fa1-8601-696a77a7aed9)

**Comment by ekzhu** [comment, marked as answer]
Have you tried to run this without fast api? 

I saw two definitions of Message type, is that intentional? 

Looks like the conversor agent was never created so the message has never arrived to it.

**Reply by IvanCord** [reply]
Hello @ekzhu,

Thank you for pointing that out. The two definitions of the `Message` type were unintentional—I was just testing the project quickly before refactoring. I changed my approach so that there’s now only one `Message` type defined in my schema folder, which I import into the relevant files (agents and the service). It works correctly now, even though the original `Message` definitions were identical. This showed me that I need a shared class for all agents that use it.

Thanks!!
